File: MyFavsDB/lib/src/models/album_search_result.dart
class AlbumSearchResult {
  final int collectionId;
  final String collectionName;
  final String artistName;
  final double collectionPrice;
  final String artworkUrl100;
  final String artworkUrl60;
  final String releaseDate;
  final int trackCount;
  final String primaryGenreName;
  final String collectionExplicitness;
  final String country;
  final String currency;
  final String collectionViewUrl;
  final String collectionCensoredName;
  final String artistViewUrl;
  final int artistId;
  final int? amgArtistId;
  final String copyright;
  final String description;

  AlbumSearchResult({
    required this.collectionId,
    required this.collectionName,
    required this.artistName,
    required this.collectionPrice,
    required this.artworkUrl100,
    required this.artworkUrl60,
    required this.releaseDate,
    required this.trackCount,
    required this.primaryGenreName,
    required this.collectionExplicitness,
    required this.country,
    required this.currency,
    required this.collectionViewUrl,
    required this.collectionCensoredName,
    required this.artistViewUrl,
    required this.artistId,
    this.amgArtistId,
    required this.copyright,
    required this.description,
  });

  factory AlbumSearchResult.fromMap(Map<String, dynamic> map) {
    return AlbumSearchResult(
      collectionId: map['collectionId'] ?? 0,
      collectionName: map['collectionName'] ?? '',
      artistName: map['artistName'] ?? '',
      collectionPrice: (map['collectionPrice'] ?? 0.0).toDouble(),
      artworkUrl100: map['artworkUrl100'] ?? '',
      artworkUrl60: map['artworkUrl60'] ?? '',
      releaseDate: map['releaseDate'] ?? '',
      trackCount: map['trackCount'] ?? 0,
      primaryGenreName: map['primaryGenreName'] ?? '',
      collectionExplicitness: map['collectionExplicitness'] ?? '',
      country: map['country'] ?? '',
      currency: map['currency'] ?? '',
      collectionViewUrl: map['collectionViewUrl'] ?? '',
      collectionCensoredName: map['collectionCensoredName'] ?? '',
      artistViewUrl: map['artistViewUrl'] ?? '',
      artistId: map['artistId'] ?? 0,
      amgArtistId: map['amgArtistId'],
      copyright: map['copyright'] ?? '',
      description: map['description'] ?? '',
    );
  }

  String get fullImageUrl {
    if (artworkUrl100.isNotEmpty) {
      return artworkUrl100.replaceAll('100x100', '300x300');
    }
    return '';
  }

  String get displayPrice {
    if (collectionPrice > 0) {
      return '${collectionPrice.toStringAsFixed(2)} $currency';
    }
    return 'Gratuito';
  }

  String get displayReleaseDate {
    if (releaseDate.length >= 4) {
      return releaseDate.substring(0, 4);
    }
    return '';
  }

  String get displayTrackCount {
    return '$trackCount faixa${trackCount > 1 ? 's' : ''}';
  }
} 

File: backend/src/main/java/com/myfavsdb/dto/AlbumSearchResult.java
package com.myfavsdb.dto;

import com.fasterxml.jackson.annotation.JsonProperty;

public class AlbumSearchResult {
    @JsonProperty("collectionId") public Long collectionId;
    @JsonProperty("collectionName") public String collectionName;
    @JsonProperty("artistName") public String artistName;
    @JsonProperty("collectionPrice") public Double collectionPrice;
    @JsonProperty("artworkUrl100") public String artworkUrl100;
    @JsonProperty("artworkUrl60") public String artworkUrl60;
    @JsonProperty("releaseDate") public String releaseDate;
    @JsonProperty("trackCount") public Integer trackCount;
    @JsonProperty("primaryGenreName") public String primaryGenreName;
    @JsonProperty("collectionExplicitness") public String collectionExplicitness;
    @JsonProperty("country") public String country;
    @JsonProperty("currency") public String currency;
    @JsonProperty("collectionViewUrl") public String collectionViewUrl;
    @JsonProperty("collectionCensoredName") public String collectionCensoredName;
    @JsonProperty("artistViewUrl") public String artistViewUrl;
    @JsonProperty("artistId") public Long artistId;
    @JsonProperty("amgArtistId") public Long amgArtistId;
    @JsonProperty("copyright") public String copyright;
    @JsonProperty("description") public String description;

    public String getFullImageUrl() {
        if (artworkUrl100 != null && !artworkUrl100.isEmpty()) {
            // Substituir 100x100 por 300x300 para melhor qualidade
            return artworkUrl100.replace("100x100", "300x300");
        }
        return null;
    }

    public String getDisplayPrice() {
        if (collectionPrice != null && collectionPrice > 0) {
            return String.format("%.2f %s", collectionPrice, currency != null ? currency : "USD");
        }
        return "Gratuito";
    }

    public String getDisplayReleaseDate() {
        if (releaseDate != null && releaseDate.length() >= 4) {
            return releaseDate.substring(0, 4); // Apenas o ano
        }
        return "";
    }

    public String getDisplayTrackCount() {
        if (trackCount != null) {
            return trackCount + " faixa" + (trackCount > 1 ? "s" : "");
        }
        return "";
    }
} 